### ISSUE TITLE
Bosh 271.10.0 Breaks ERB templates

### ISSUE THREAD

**Issue by ChrisMcGowan** [cloudfoundry/bosh#2322]
**Describe the bug**
Upgraded director from `271.9.0` to `271.10.0`- during a CF deployment the ERB rendering of the `gorouter.yml` file by the director adds erroneous lines to the `gorouter.yml` file causing it to break

**To Reproduce**
Steps to reproduce the behavior (example):
1. Deploy a bosh director on AWS to using version `271.9.0`
2. Deploy a normal/plan  `cf-deployment` 
3. Upgrade director to `271.10.1`
4. Re-deploy `cf-deployment` with no changes and you will see the gorouter VMs get updated and fail

**Expected behavior**
That ERB template rendering behaves the same between director releases

**Logs**
Nothing really on director that shows any errors/seems to behave as normal

**Versions (please complete the following information):**
 - Infrastructure: AWS
 - BOSH version 271.10.0
 - BOSH CLI version 6.1.1
 - Stemcell version Ubuntu-bionic 1.28
 - CF-Deployment - 16.23
 - Routing Release 0.223.0

**Additional context**

Default gorouter job values of an array of 4 credhub certs to be used: https://github.com/cloudfoundry/cf-deployment/blob/main/cf-deployment.yml#L1366-L1370

Gorouter erb templete for `gorouter.yml` : https://github.com/cloudfoundry/routing-release/blob/0.223.0/jobs/gorouter/templates/gorouter.yml.erb#L295-L315

Using director `271.10.1` the gorouter.yml fields render like this:

```
ca_certs: |+
<cert 1>
<cert 2>
<cert 3>
<cert4>
<blank line>
...
<5 blank lines>
client_ca_certs: |+
<cert 1>
<cert 2>
<cert 3>
<cert4>
<blank line>
...
<3 blank lines>
only_trust_client_ca_certs
```

It appears this type of templating is also used in the routing_api as well but we don't use that function.

**Temp Fix**
Rolled back Director to `271.9.0` and pinned `bosh-deployment` resource for now

**Comment by ChrisMcGowan** [comment]
Quick add:

`gorouter.yml` under Bosh `271.9.0`

```
ca_certs: |+
<cert 1>
<cert 2>
<cert 3>
<cert 4>
<6 blank lines>
client_ca_certs: |+
<cert 1>
<cert 2>
<cert 3>
<cert 4>
<4 blank lines>
only_trust_client_ca_certs
```

**Comment by rkoster** [comment]
The issue is related to new behaviour in libyaml. This behaviour has been removed via a patch here: https://github.com/bosh-packages/ruby-release/pull/23. The next director will ship this change, and should be compatible with the routing-release again.

**Comment by jpalermo** [comment]
Should be fixed in 271.11.0.

Thanks for opening the issue!
